js/components/generate/GenerateForm.js

```diff
@@ -296,91 +298,116 @@ export default class GenerateForm extends React.Component {
-                    step="1"
-                    style={{ marginBottom: "1em" }}
-                  />
-                </p>
-
-                {/* <label htmlFor="location">Campus</label>
-                <Field id="location" name="location" as="select" multiple
-                  style={{ verticalAlign: 'text-top', marginLeft: '1em', marginBottom: '1em', color: 'black' }}>
-                  <option value="utsg">St. George</option>
-                  <option value="utm">Mississauga</option>
-                  <option value="utsc">Scarborough</option>
-                </Field>
-
-                <p>
-                  <label htmlFor="includeRaws">Include non-course prerequisites</label>
-                  <Field id="includeRaws" name="includeRaws" type="checkbox"
-                    style={{ marginLeft: '1em', verticalAlign: 'middle' }}
+                    id="courses"
+                    name="courses"
+                    type="text"
+                    placeholder="e.g., CSC207H1, CSC324H1"
                   />
-                </p>
+                  <div className="error-container">
+                    <ErrorMessage
+                      className="error-message"
+                      name="courses"
+                      component="div"
+                    />
+                  </div>
+                </div>
 
-                <label htmlFor="includeGrades">Include grade-based prerequisites</label>
-                <Field id="includeGrades" name="includeGrades" type="checkbox"
-                  style={{ 'margin-left': '1em', 'vertical-align': 'middle' }} /> */}
+                <div className="form-section">
+                  <h2 id="filter-title" className="section-title">Filters</h2>
+
+                  <div className="title-container">
+                    <label htmlFor="departments">Departments</label>
+                    <a
+                      data-tooltip-id="departments-tooltip"
+                      data-tooltip-content="Only include courses from these departments"
+                      className="tooltip-icon">
+                    </a>
+                    <Tooltip id="departments-tooltip" place="right" />
+                  </div>
+                  <Field
+                    id="departments"
+                    name="departments"
+                    type="text"
+                    placeholder="Enter 3 letter dept. codes, sep. by commas"
+                  />
+                  <div className="error-container">
+                    <ErrorMessage
+                      className="error-message"
+                      name="departments"
+                      component="div"
+                    />
+                  </div>
+
+                  <div className="title-container">
+                    <label htmlFor="taken">Hide courses</label>
+                    <a
+                      data-tooltip-id="taken-tooltip"
+                      data-tooltip-content="Do not show these courses"
```

Comment:
Add "or their prerequisites" to the end of this description. Also specify the required format, similar to the "Search Course(s)" help field.